jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -425,13 +425,13 @@ public void matchCreated(MatchCreatedEvent event) {
                     // new activations of the rule associate with a state node
                     // signal process instances of that state node
                     String ruleName = event.getMatch().getRule().getName();
-                    if ( ruleName.startsWith( "RuleFlowStateNode-" )) {
-                        int index = ruleName.indexOf( '-',
-                                                      18 );
-                        index = ruleName.indexOf( '-',
-                                                  index + 1 );
-                        String eventType = ruleName.substring( 0,
-                                                               index );
+                    String rulePrefix = "RuleFlowStateNode-";
+                    if (ruleName.startsWith(rulePrefix)) {
+                        int index = ruleName.indexOf('-', rulePrefix.length() + 1);
```

Comment:
This is not used at all since this is overwritten on line 433.

Can you also please further describe why you have to remove last 2 hyphens? Thanks.